tests/integration/test_error_injection.py

```diff
@@ -1,117 +1,109 @@
-    assert test_no_errors['ResponseMetadata']['HTTPStatusCode'] == 200
+    @classmethod
+    def setUpClass(cls) -> None:
+        if not should_run():
+            pytest.skip('skipping TestErrorInjection (TEST_ERROR_INJECTION not set or false)')
 
+    def test_kinesis_error_injection(self):
+        kinesis = aws_stack.connect_to_service('kinesis')
+        aws_stack.create_kinesis_stream(TEST_STREAM_NAME)
 
-def test_dynamodb_error_injection():
-    if not do_run():
-        return
+        records = [
+            {
+                'Data': '0',
+                'ExplicitHashKey': '0',
+                'PartitionKey': '0'
+            }
+        ]
 
-    table = get_dynamodb_table()
+        # by default, no errors
+        test_no_errors = kinesis.put_records(StreamName=TEST_STREAM_NAME, Records=records)
+        assert test_no_errors['FailedRecordCount'] == 0
 
-    partition_key = short_uid()
-    assert_zero_probability_read_error_injection(table, partition_key)
+        # with a probability of 1, always throw errors
+        config.KINESIS_ERROR_PROBABILITY = 1.0
+        test_all_errors = kinesis.put_records(StreamName=TEST_STREAM_NAME, Records=records)
+        assert test_all_errors['FailedRecordCount'] == 1
 
-    # with a probability of 1, always throw errors
-    config.DYNAMODB_ERROR_PROBABILITY = 1.0
-    with pytest.raises(ClientError):
-        table.get_item(Key={PARTITION_KEY: partition_key})
+        # reset probability to zero
+        config.KINESIS_ERROR_PROBABILITY = 0.0
 
-    # reset probability to zero
-    config.DYNAMODB_ERROR_PROBABILITY = 0.0
+    def get_dynamodb_table(self):
+        dynamodb = aws_stack.connect_to_resource('dynamodb')
+        # create table with stream forwarding config
+        aws_stack.create_dynamodb_table(TEST_TABLE_NAME, partition_key=PARTITION_KEY)
+        return dynamodb.Table(TEST_TABLE_NAME)
 
+    def assert_zero_probability_read_error_injection(self, table, partition_key):
+        # by default, no errors
+        test_no_errors = table.get_item(Key={PARTITION_KEY: partition_key})
+        assert test_no_errors['ResponseMetadata']['HTTPStatusCode'] == 200
 
-def test_dynamodb_read_error_injection():
-    if not do_run():
-        return
+    def test_dynamodb_error_injection(self):
 
-    table = get_dynamodb_table()
+        table = self.get_dynamodb_table()
 
-    partition_key = short_uid()
-    assert_zero_probability_read_error_injection(table, partition_key)
+        partition_key = short_uid()
+        self.assert_zero_probability_read_error_injection(table, partition_key)
 
-    # with a probability of 1, always throw errors
-    config.DYNAMODB_READ_ERROR_PROBABILITY = 1.0
-    with pytest.raises(ClientError):
-        table.get_item(Key={PARTITION_KEY: partition_key})
+        # with a probability of 1, always throw errors
+        config.DYNAMODB_ERROR_PROBABILITY = 1.0
+        with pytest.raises(ClientError):
```

Comment:
nit: we could also use the `unittest` variant of this (not critical, shouldn't hold back the merge):
```
with self.assertRaises(ClientError):
    ...
```

Reply:
good point! much cleaner